compact_str/src/features/sqlx.rs

```diff
@@ -0,0 +1,113 @@
+use sqlx::database::HasValueRef;
+use sqlx::error::BoxDynError;
+#[cfg(any(
+    feature = "sqlx-mysql",
+    feature = "sqlx-postgres",
+    feature = "sqlx-sqlite"
+))]
+use sqlx::{
+    database::HasArguments,
+    encode::IsNull,
+    Encode,
+};
+use sqlx::{
+    Database,
+    Decode,
+    Type,
+    Value,
+    ValueRef,
+};
+
+use crate::{
+    CompactString,
+    ToCompactString,
+};
+
+#[cfg_attr(docsrs, doc(cfg(feature = "sqlx")))]
+impl<DB> Type<DB> for CompactString
+where
+    DB: Database,
+    for<'x> &'x str: Type<DB>,
+{
+    #[inline]
+    fn type_info() -> <DB as Database>::TypeInfo {
+        <&str as Type<DB>>::type_info()
+    }
+}
+
+#[cfg_attr(docsrs, doc(cfg(feature = "sqlx")))]
+impl<'r, DB> Decode<'r, DB> for CompactString
+where
+    DB: Database,
+    for<'x> &'x str: Decode<'x, DB> + Type<DB>,
+{
+    fn decode(value: <DB as HasValueRef<'r>>::ValueRef) -> Result<Self, BoxDynError> {
+        let value = value.to_owned();
+        let value: &str = value.try_decode()?;
+        Ok(value.try_to_compact_string()?)
+    }
+}
```

Comment:
Is there a reason to not specifically implement Decode for each of the Database variants? They each look like they have faster code paths for decoding `&str` that can be used to create a CompactStr.